server/controllers/interview/interviewController.ts

```diff
@@ -0,0 +1,72 @@
+import { Response } from 'express';
+import Interviewee from '../../database/Models/Interviewee';
+import User from '../../database/Models/User';
+import { CustomError, mailSender, RequestType } from '../../utils';
+import interviewValidation from '../../utils/validation/interviewValidation';
+import emailTemplate from './interviewEmailTemplate';
+
+const createInterview = async (req: RequestType, res: Response) => {
+  const id = req.userInfo?.id;
+  const {
+    interviewerId, date, time, language, specialization, questionCategory,
+  } = req.body;
+
+  // Validate the incoming request
+  await interviewValidation(req.body);
+
+  // Get the interviewee and interviewer
+  const { email: intervieweeEmail } = await User.findById(id);
+  const { email: interviewerEmail } = await User.findById(interviewerId);
+
+  // Get the interviewee interviews
+  const { interviews } = await Interviewee.findOne({
+    where: {
+      userId: id,
+    },
+  });
+
+  const interview = {
+    interviewerId,
+    date,
+    time,
+    language,
+    specialization,
+    questionCategory,
+  };
+
+  // Add the interview to the interviewee interviews
+  if (interviews) {
+    interviews.push(interview);
+  }
+
+  // Update the interviewee interviews
+  const updatedInterviews = await Interviewee.updateOne({
+    where: {
+      userId: id,
+    },
+    interviews,
+  });
```

Comment:
Why do you find first and then update ? why not update immediately?
Before updating you need to check if the date and the time are available so we don't have a contradiction.
And we have two collections that contain interviews so we need to update both of them